src/charm.py

```diff
@@ -111,7 +109,30 @@ def _load_nginx_config(self) -> None:
             )
             status_message = f"Error for host: {err.hosts}"
 
-        self.unit.status = ops.ActiveStatus(status_message)
+        for _ in range(6):
+            self._update_status_with_nginx()
+            if isinstance(self.unit.status, ops.ActiveStatus):
+                self.unit.status = ops.ActiveStatus(status_message)
+                break
+            sleep(5)
+
+    def get_config_and_update_status(self) -> NginxConfig | None:
+        """Attempt to get nginx config, updates charm status on failure.
+
+        Returns:
+            The nginx configuration if found and valid.
+        """
+        try:
+            nginx_config = get_nginx_config(self)
+        except IntegrationDataError as err:
+            self.unit.status = ops.BlockedStatus(str(err))
+            return None
+        if not nginx_config:
+            _nginx_stop()
```

Comment:
Should we stop nginx in a method named `get_config_and_update_status` ? 
This behavior is not one I'd expect, and the docstring doesn't describe this behavior